### PR TITLE
Adding TLS

### PR DESCRIPTION
**Pull request by jonathankingfc**
**Issue:** https://issues.redhat.com/browse/PROJQUAY-???

**Changelog:** 
- TLS has been added to config-tool

**Docs:** 
- A user should add the following flags to their Docker run command in order to configure TLS.

	-v ./testdata-tls/localhost.crt:${CONFIG_TOOL_PUBLIC_KEY} \
	-v ./testdata-tls/localhost.key:${CONFIG_TOOL_PRIVATE_KEY} \
	-e CONFIG_TOOL_PUBLIC_KEY=${CONFIG_TOOL_PUBLIC_KEY} \
	-e CONFIG_TOOL_PRIVATE_KEY=${CONFIG_TOOL_PRIVATE_KEY} 

First, we give the container access to your TLS files through a mounted volume. Then, we specify the internal mount location.

**Testing:** 

**Details:** 

------

### CODE REVIEW

File: pkg/lib/editor/editor.go [hunk 1]
```diff
@@ -111,11 +121,49 @@ func RunConfigEditor(password, configPath, operatorEndpoint string, readOnlyFiel
 		fs.ServeHTTP(w, r)
 	})
 
+	// Create server base
 	s := &http.Server{
 		Addr:    ":" + strconv.Itoa(opts.port),
```
Comment (ldelossa) [inline pkg/lib/editor/editor.go:126]: should take port as a string, its even in the c libs it is used as a string.

File: pkg/lib/editor/editor.go [hunk 2]
```diff
@@ -111,11 +121,49 @@ func RunConfigEditor(password, configPath, operatorEndpoint string, readOnlyFiel
...
 		Handler: r,
 	}
 
-	log.Fatal(s.ListenAndServe())
+	// Try to load TLS
+	tlsConfig, err := loadTLS(opts.publicKeyPath, opts.privateKeyPath)
+	switch err == nil {
```
Comment (ldelossa) [inline pkg/lib/editor/editor.go:132]: Why not follow the idiom here and just do 

if err != nil  ?

File: pkg/lib/editor/editor.go [hunk 3]
```diff
@@ -111,11 +121,49 @@ func RunConfigEditor(password, configPath, operatorEndpoint string, readOnlyFiel
...
+	if err != nil {
+		return nil, errors.New("Could not open private key: " + privateKeyPath)
+	}
+	cert, err := tls.X509KeyPair(crt, key)
+	if err != nil {
+		return nil, errors.New("Could not load X509 key pair: " + err.Error())
+	}
 
+	return &tls.Config{
+		Certificates: []tls.Certificate{cert}}, nil
```
Comment (ldelossa) [inline pkg/lib/editor/editor.go:168]: For your purposes, do you also need to add the system cert pool? https://golang.org/src/crypto/x509/cert_pool.go?s=1628:1668#L49 these are the trusted root certs located on the server.

**Review by ldelossa — CHANGES_REQUESTED** [review]
couple questions

File: pkg/lib/editor/editor.go (hunk 1 above)
Comment (jonathankingfc) [reply pkg/lib/editor/editor.go:126]: it does take it as a string, but since i don't specify a hostname i need to put the ":8080"

Comment (ldelossa) [reply pkg/lib/editor/editor.go:126]: Why are you performing a string conversation from integer to string then?

Comment (jonathankingfc) [reply pkg/lib/editor/editor.go]: good point, was reading port as in int from cli. fixed.

**Review by ldelossa — APPROVED** [review]

File: pkg/lib/editor/editor.go (hunk 3 above)
Comment (jonathankingfc) [reply pkg/lib/editor/editor.go:165]: nope, not needed for this server. root certs are used at a later point but only when creating transports to external services.

**Pull request merged**